v0/destinations/yahoo_dsp/util.js

```diff
@@ -0,0 +1,283 @@
+const qs = require("qs");
+const sha256 = require("sha256");
+const { generateJWTToken } = require("../../../util/jwtTokenGenerator");
+const {  httpPOST } = require("../../../adapters/network");
+const { CustomError, isDefinedAndNotNullAndNotEmpty } = require("../../util");
+
+const { ACCESS_TOKEN_CACHE_TTL } = require("./config.js");
+const Cache = require("../../util/cache");
+
+const ACCESS_TOKEN_CACHE = new Cache(ACCESS_TOKEN_CACHE_TTL);
+
+const getUnixTimestamp = () => {
+  return Math.floor(Date.now() / 1000);
+};
+
+/**
+ * There are four location types which are stored in the poiLocationType error. These are ["chains", "woeids", "gids", "categories"].
+ * The ones to be added in includes Object will be taken from audienceType with include as Prefix. The ones to be added in excludes
+ * Object will be taken from audienceType with exclude as Prefix.
+ */
+const poiLocationType = [
+  "includeChains",
+  "includeWoeids",
+  "includeCategories",
+  "includeGids",
+  "excludeChains",
+  "excludeWoeids",
+  "excludeCategories",
+  "excludeGids"
+];
+let listType;
+const includes = {};
+const excludes = {};
+let seedList = [];
+let outputPayload = {};
+
+
+/**
+ * This function is used to check if there is any common value in keys and poiLocationType array. This is done to check
+ * if any of the required location type is present in the input or not. 
+ * @param {*} keys
+ * @param {*} poiLocationType
+ * @returns 
+ */
+function isCommonElement(keys, poiLocationType) {
+  return keys.some(item => poiLocationType.includes(item))
+}
+
+
+/**
+ *
+ * @param {*} audienceList  - It is the list of audience to be added in the form of array". eg.
+ * [
+ *   {"email": "abc@email.com"},
+ *   {"email": "abc@email.com"},
+ *   {"email": "abc@email.com"}
+ * ]
+ * @param {*} Config
+ * @returns The function returns an array of Audience List provided by the user like "email", "deviceId", "ipAddress".
+ * eg.[
+ * "251014dafc651f68edac7",
+ * "afbc34416ac6e7fbb9734",
+ * "42cbe7eebb412bbcd5b56",
+ * "379b4653a40878da7a584"
+ * ]
+ */
+const populateIdentifiers = (audienceList, Config) => {
+  const seedList = [];
+  const { audienceType } = Config;
+  const { hashRequired } = Config;
+  let listType;
+  if (isDefinedAndNotNullAndNotEmpty(audienceList)) {
+    // traversing through every userTraits in the add array for the traits to be added.
+    audienceList.forEach(userTraits => {
+      // storing keys of an object inside the add array.
+      const keys = Object.keys(userTraits);
+      // checking for the audience type the user wants to add is present in the input or not.
+      if(!keys.includes(audienceType)){
```

Comment:
Use an enum for this not directly a config to parse data
example

Store this in config.js
```
const AUDIENCE_TYPE_ENUM = {
    EMAIL:  "email"
    DEVICE_ID: "device_id"
    ...
}
```

During the check it can be done as 

```
if(!AUDIENCE_TYPE_ENUM[audienceType]) {
    .. throw error
}
```

You need to change audienceType value in webapp